### PR TITLE
Add io.github.mhogomchungu.media-downloader

### PR DESCRIPTION
**Pull request by mhogomchungu**
### Please confirm your submission meets all the criteria

- [X]` Please describe the application briefly.
  This application is a Qt based front end to [yt-dlp](https://github.com/yt-dlp/yt-dlp), [gallery-dl](https://github.com/mikf/gallery-dl), [lux](https://github.com/iawia002/lux) and [svtplay-dl](https://svtplay-dl.se/). These are CLI tools and they deal with downloading of audio files, video files and image files from many websites.

- [X]` The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X]` I have read the and followed all the [Submission and licence requirements][reqs].
- [X]`  I have [built][build] and tested the submission locally.
- [X]` I am the developer of the project.

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.mhogomchungu.media-downloader.xml
Comment (hfiguiere) [inline io.github.mhogomchungu.media-downloader.xml:1]: this file is never referenced anywhere.

Comment (mhogomchungu) [reply io.github.mhogomchungu.media-downloader.xml:1]: Where should it me mentioned and how?

It is not found in  the [manifest example](https://docs.flatpak.org/en/latest/manifests.html) and i have not seen any examples of how to incorporate it in the manifest file.

The file is found in the [archive](https://github.com/mhogomchungu/media-downloader/tree/main/src/flatpak) and it gets installed by cmake together with other files during install time.

```
-- Install configuration: "release"
-- Installing: /app/bin/io.github.mhogomchungu.media-downloader
-- Installing: /app/share/metainfo/io.github.mhogomchungu.media-downloader.metainfo.xml
-- Installing: /app/share/icons/hicolor/512x512/apps/io.github.mhogomchungu.media-downloader.png
-- Installing: /app/share/applications/io.github.mhogomchungu.media-downloader.desktop
-- Installing: /app/share/media-downloader/translations
-- Installing: /app/share/media-downloader/translations/fr_FR.qm
-- Installing: /app/share/media-downloader/translations/en_US.qm
-- Installing: /app/share/media-downloader/translations/tr_TR.qm
-- Installing: /app/share/media-downloader/translations/pl_PL.qm
-- Installing: /app/share/media-downloader/translations/de_DE.qm
-- Installing: /app/share/media-downloader/translations/sv_SE.qm
-- Installing: /app/share/media-downloader/translations/ar.qm
-- Installing: /app/share/media-downloader/translations/zh_CN.qm
-- Installing: /app/share/media-downloader/translations/nl_NL.qm
-- Installing: /app/share/media-downloader/translations/pt_BR.qm
-- Installing: /app/share/media-downloader/translations/ko_KR.qm
-- Installing: /app/share/media-downloader/translations/ru_RU.qm
-- Installing: /app/share/media-downloader/translations/it_IT.qm
-- Installing: /app/share/media-downloader/translations/es.qm
-- Installing: /app/share/media-downloader/translations/ja_JP.qm

```

File: io.github.mhogomchungu.media-downloader.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        "--socket=wayland",
+        "--socket=fallback-x11",
+        "--filesystem=xdg-download",
+        "--share=network",
+        "--device=dri"
+    ],
+    "modules": [
+        {
+            "name": "io.github.mhogomchungu.media-downloader",
+            "buildsystem": "cmake",
```
[the remark this answers is not in the document]
Comment (mhogomchungu) [reply io.github.mhogomchungu.media-downloader.json]: Done.

File: io.github.mhogomchungu.media-downloader.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	<metadata_license>FSFAP</metadata_license>
+
+	<project_license>GPL-2.0-or-later</project_license>
+
+	<url  type="homepage">https://mhogomchungu.github.io/media-downloader</url>
+	<url type="bugtracker">https://github.com/mhogomchungu/media-downloader/issues</url>
+
+	<content_rating type="oars-1.1" />
+
+	<developer id="io.github.mhogomchungu.media-downloader">
```
Comment (bbhtt) [inline io.github.mhogomchungu.media-downloader.metainfo.xml:19]:
```suggestion
	<developer id="io.github.mhogomchungu">
```

File: io.github.mhogomchungu.media-downloader.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		  <name>Francis Banyikwa</name>
+	</developer>
+
+	<description><p>This application is a front end to command line tools that deals with downloading of online media and can be used to download video files, audio files and images from many website.</p></description>
+
+	<icon>io.github.mhogomchungu.media-downloader</icon>
```
Comment (bbhtt) [inline io.github.mhogomchungu.media-downloader.metainfo.xml:25]:
```suggestion
```

File: io.github.mhogomchungu.media-downloader.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	</categories>
+
+	<releases>
+		<release version="4.9.0" type="stable" date="2024-08-01" />
+		<release version="4.8.0" type="stable" date="2024-07-03" />
+	</releases>
+
+	<provides>
+		<binary>io.github.mhogomchungu.media-downloader</binary>
+	</provides>
```
Comment (bbhtt) [inline io.github.mhogomchungu.media-downloader.metainfo.xml:53]:
```suggestion
```

File: io.github.mhogomchungu.media-downloader.metainfo.xml
Comment (bbhtt) [inline io.github.mhogomchungu.media-downloader.metainfo.xml:1]: This file should be upstream

File: io.github.mhogomchungu.media-downloader.xml
Comment (bbhtt) [reply io.github.mhogomchungu.media-downloader.xml:1]: > The file is found in the [archive](https://github.com/mhogomchungu/media-downloader/tree/main/src/flatpak)

Then why is a duplicate copy here? Apply suggestions, update the one in archive and drop this.

**Comment by mhogomchungu** [comment]
Done making recommended changes.

All files are now found upstream [here](https://github.com/mhogomchungu/media-downloader/tree/main/src/flatpak).

**Comment by bbhtt** [comment]
> All files are now found upstream [here](https://github.com/mhogomchungu/media-downloader/tree/main/src/flatpak).

You removed the manifest too. The manifest needs to be here.

**Comment by mhogomchungu** [comment]
Done.

Manifest file is back and it has links to upstream desktop ,metainfo and icon files.

**Comment by bbhtt** [comment]
bot, build io.github.mhogomchungu.media-downloader

**Comment by mhogomchungu** [comment]
Changes already made that corrected problems that caused the build to fail.

**Comment by mhogomchungu** [comment]
bot, build io.github.mhogomchungu.media-downloader

**Comment by mhogomchungu** [comment]
bot, build io.github.mhogomchungu.media-downloader

**Comment by mhogomchungu** [comment]
bot, build io.github.mhogomchungu.media-downloader

**Comment by mhogomchungu** [comment]
bot, build io.github.mhogomchungu.media-downloader

**Comment by mhogomchungu** [comment]
The build above works as expected

Screenshot_20240806_235147

**Comment by mhogomchungu** [comment]
bot, build io.github.mhogomchungu.media-downloader

**Pull request closed without merging**